README.md

```diff
@@ -137,7 +140,106 @@ function debug(plugins) {
 }
 ```
 
-##### `lint`
+#### `encapsulate`
+
+* Type: `Boolean`
+* Default: `false`
+
+Resets CSS properties to their default values to effectively allow a component
+to opt out of CSS inheritance and be encapsulated from the rest of the
+application similar to [the Shadow DOM](https://www.w3.org/TR/shadow-dom/).
+There are two types of CSS properties that affect components, inherited (e.g.
+`font-size,` `color`) and non-inherited (e.g. `margin`, `background`). This
+option works so that:
+
+* Root elements (e.g. `.Component`) have both inherited and non-inherited
+  properties reset to default values.
+* Descendants (e.g. `.Component-item`) only have non-inherited properties reset
+  as this allows properties set on the root element to be inherited by its
+  descendants.
+
+This means that components are isolated from styles outside the component root
+element but should an inheritable property such as `font-size` be applied on the
+component root element it will be inherited by the component descendants as
+normal. The same rules apply to nested components.
+
+##### Why not use `all: initial`?
+
+The `all: initial` declaration will reset even the User Agent styles which is
+often too heavy handed:
+
+> For example, if an author specifies `all: initial` on an element it will block
+  all inheritance and reset all properties, as if no rules appeared in the
+  author, user, or user-agent levels of the cascade.
+  This can be useful for the root element of a "widget" included in a page, which
+  does not wish to inherit the styles of the outer page. Note, however, that any
+  "default" style applied to that element (such as, e.g. `display: block` from the
+  UA style sheet on block elements such as `<div>`) will also be blown away.
```

Comment:
> any "default" style applied to that element will also be blown away

In a way this is also true in our case because we are using a common reset for every element. 
So we can't reset to UA styles based on the type of tag because we are using class selectors.
For example `a` tags reset by us won't be "blue" and have `text-decoration: underline`.

The only difference is that `all: initial` resets both all inherited and not inherited properties, while we picked a subset of properties and left out some like `display`.